### ISSUE TITLE
Make `ExperimentSpec` work with `ValueFunction` not Q tables only

### ISSUE THREAD

**Issue by wasowski** [itu-square/symsim#135]

**Comment by wasowski** [comment]
@mohsen-ghaffari1992 , found this comment from you somewhere in ConcreteValueTable:
```
  /** is the following part necessary for Value Table?! */
//  def runV: V =
//    val initials = Randomized.repeat (agent.initialize).take (episodes)
//    val schedule = learn (this.initialize, initials)
//    schedule.head
//
//  override def run: Policy =
//    qToPolicy (this.runQ)
//
//  /** Convert the matrix Q after training into a Policy map. TODO: should not
//    * this be using the bestAction method? Or, why is the best action method
//    * abstract? Or is qToPolicy too concrete to be here?
//    */
//  def qToPolicy (q: Q) (using Ordering[Double]): Policy =
//    def best (m: Map[Action, Double]): Action =
//      m.map { _.swap } (m.values.max)
//    q.view.mapValues (best).to (Map)
//
//  /** We assume that all values define the same set of actions valuations.  */
//  def pp_Q (f: Q): Doc =
//    val headings = "" ::f
//            .values
//            .head
//            .keys
//            .map (_.toString)
//            .toList
//            .sorted
//    def fmt (s: ObservableState, m: Map[Action, Double]): List[String] =
//      s.toString ::m
//              .toList
//              .sortBy (_._1.toString)
//              .map { _._2.toString.take (7).padTo (7,'0') }
//    val rows = f
//            .toList
//            .sortBy (_._1.toString)
//            .map (fmt)
//    symsim.tabulate (' ', " | ", headings ::rows, "-".some, "-+-".some)
```
This is exactly what this issue is about :). The answer is yes. Otherwise you will not be able to run experiments with value-table-based algorithms.

**Comment by mohsen-ghaffari1992** [comment]
Yes, those are defined for Q-Table, but I wasn’t quite sure that is necessary for Value Table also.

From: Andrzej Wąsowski ***@***.***>
Date: Wednesday, 28 September 2022 at 16.02
To: itu-square/symsim ***@***.***>
Cc: Mohsen Ghaffari ***@***.***>, Mention ***@***.***>
Subject: Re: [itu-square/symsim] Make `ExperimentSpec` work with `ValueFunction` not Q tables only (Issue #135)

@mohsen-ghaffari1992<https://github.com/mohsen-ghaffari1992> , found this comment from you somewhere in ConcreteValueTable:

  /** is the following part necessary for Value Table?! */

//  def runV: V =

//    val initials = Randomized.repeat (agent.initialize).take (episodes)

//    val schedule = learn (this.initialize, initials)

//    schedule.head

//

//  override def run: Policy =

//    qToPolicy (this.runQ)

//

//  /** Convert the matrix Q after training into a Policy map. TODO: should not

//    * this be using the bestAction method? Or, why is the best action method

//    * abstract? Or is qToPolicy too concrete to be here?

//    */

//  def qToPolicy (q: Q) (using Ordering[Double]): Policy =

//    def best (m: Map[Action, Double]): Action =

//      m.map { _.swap } (m.values.max)

//    q.view.mapValues (best).to (Map)

//

//  /** We assume that all values define the same set of actions valuations.  */

//  def pp_Q (f: Q): Doc =

//    val headings = "" ::f

//            .values

//            .head

//            .keys

//            .map (_.toString)

//            .toList

//            .sorted

//    def fmt (s: ObservableState, m: Map[Action, Double]): List[String] =

//      s.toString ::m

//              .toList

//              .sortBy (_._1.toString)

//              .map { _._2.toString.take (7).padTo (7,'0') }

//    val rows = f

//            .toList

//            .sortBy (_._1.toString)

//            .map (fmt)

//    symsim.tabulate (' ', " | ", headings ::rows, "-".some, "-+-".some)

This is exactly what this issue is about :). The answer is yes. Otherwise you will not be able to run experiments with value-table-based algorithms.

**Comment by wasowski** [comment]
Another tip: do not respond to PR/issue discussions by email because then responses are very hard to read on github.com :)
